core/src/main/scala/org/apache/spark/Dependency.scala

```diff
@@ -206,6 +206,20 @@ class ShuffleDependency[K: ClassTag, V: ClassTag, C: ClassTag](
     finalizeTask = Option(task)
   }
 
+  private val numberOfShuffleBlocks = numPartitions.toLong * partitioner.numPartitions.toLong
```

Comment:
Please see [discussion here](https://github.com/apache/spark/pull/45266#discussion_r1508350378) - let us avoid introducing a new field in `ShuffleDependency`.